src/WorkflowDefinition.php

```diff
@@ -104,6 +104,23 @@ public function addWorkflow(AbstractWorkflow $workflow, array $dependencies = []
         return $this;
     }
 
+    /**
+     * @param bool|callable(): bool $condition
+     * @param callable(WorkflowDefinition): void $callback
+     */
+    public function when(callable|bool $condition, callable $callback): self
```

Comment:
Couldn't we use the `Conditionable` trait in `Illuminate\Support` for this feature?

Reply:
Oh, I didn't even know about this trait. That's probably a better idea.

That's why I pay you the big (exposure) bucks!